Tiltfile

```diff
@@ -0,0 +1,145 @@
+    chmod +x /start.sh && chmod +x /restart.sh
+
+FROM debian:stretch-slim as tilt
+WORKDIR /
+COPY --from=tilt-helper /start.sh .
+COPY --from=tilt-helper /restart.sh .
+COPY .tiltbuild/manager .
+ENTRYPOINT [ "/start.sh", "/manager" ]
+"""
+
+# Configures a provider by doing the following:
+#
+# 1. Enables a local_resource go build of the provider's manager binary
+# 2. Configures a docker build for the provider, with live updating of the manager binary
+# 3. Runs kustomize for the provider's config/default and applies it
+def enable_provider(name):
+    p = providers.get(name)
+
+    context = p.get('context')
+
+    # Prefix each live reload dependency with context. For example, for if the context is
+    # test/infra/docker and main.go is listed as a dep, the result is test/infra/docker/main.go. This adjustment is
+    # needed so Tilt can watch the correct paths for changes.
+    live_reload_deps=[]
+    for d in p.get('live_reload_deps', []):
+        live_reload_deps += [context + '/' + d]
+
+    # Set up a local_resource build of the provider's manager binary. The provider is expected to have a main.go in
+    # manager_build_path. The binary is written to .tiltbuild/manager.
+    local_resource(name + "_manager",
+                   cmd='cd ' + context + ';mkdir -p .tiltbuild;CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -a -ldflags \'-extldflags "-static"\' -o .tiltbuild/manager',
+                   deps=live_reload_deps)
+
+    # Set up an image build for the provider. The live update configuration syncs the output from the local_resource
+    # build into the container.
+    docker_build(
+        ref=p.get('image'),
+        context=p.get('context'),
+        dockerfile_contents=tilt_dockerfile,
+        target='tilt',
+        entrypoint='/start.sh /manager',
+        only='.tiltbuild/manager',
+        live_update=[
+            sync(context + '/.tiltbuild/manager', '/manager'),
+            run('/restart.sh')
+        ]
+    )
+
+    # Apply the kustomized yaml for this provider
+    yaml = str(kustomize(context + '/config/default'))
+    substitutions = settings.get('kustomize_substitutions', {})
+    for substitution in substitutions:
+        value = substitutions[substitution]
+        yaml = yaml.replace("${" + substitution + "}", value)
+    k8s_yaml(blob(yaml))
+
+
+# Prepull all the cert-manager images to your local environment and then load them directly into kind. This speeds up
+# setup if you're repeatedly destroying and recreating your kind cluster, as it doesn't have to pull the images over
+# the network each time.
+def deploy_cert_manager():
+    registry = 'quay.io/jetstack'
+    version = 'v0.11.0'
+    images = ['cert-manager-controller', 'cert-manager-cainjector', 'cert-manager-webhook']
+
+    for image in images:
+        local('docker pull {}/{}:{}'.format(registry, image, version))
+        local('kind load docker-image {}/{}:{}'.format(registry, image, version))
```

Comment:
Assumption here about using `kind`, which tripped up my setup, though happy to make it agnostic in post.

Reply:
+1 to follow-up